### PR TITLE
Add `ruby-lsp-doctor` tool for troubleshooting indexing problems

### PR DESCRIPTION
**Pull request by andyw8**
### Motivation

We have a script for troubleshooting issues related to indexing, but it's sensitive to the version of Ruby LSP is use, so https://github.com/Shopify/ruby-lsp/issues/1115 had a lot of back-and-forward. To make things easier in future, let's ship a command people can easily run.

### Implementation

Adapted from https://github.com/Shopify/ruby-lsp/issues/1063#issuecomment-1749018562

### Automated Tests

I added a step to CI.

### Manual Tests

Run `./exe/ruby-lsp-index-info`.

### CODE REVIEW

File: ruby-lsp.gemspec [hunk 1]
```diff
@@ -14,7 +14,7 @@ Gem::Specification.new do |s|
 
   s.files = Dir.glob("lib/**/*.rb") + ["README.md", "VERSION", "LICENSE.txt"]
   s.bindir = "exe"
-  s.executables = ["ruby-lsp", "ruby-lsp-check"]
+  s.executables = ["ruby-lsp", "ruby-lsp-check", "ruby-lsp-index-info"]
```
Comment (andyw8) [inline ruby-lsp.gemspec:17]: What might be a better name for this?

Comment (vinistock) [reply ruby-lsp.gemspec:17]: How about `ruby-lsp-debug`?

Comment (andyw8) [reply ruby-lsp.gemspec:17]: How about `ruby-lsp-doctor` (like `brew doctor`) to avoid any confusion with the `debug` gem?

Comment (andyw8) [reply ruby-lsp.gemspec:17]: My thought was that it might become a more general purpose tool for troubleshooting setups in future.

Comment (vinistock) [reply ruby-lsp.gemspec:17]: Yeah, I agree with that. We may want to expand that for other functionality and renaming the executable later may prove painful.

Comment (st0012) [reply ruby-lsp.gemspec:17]: I understand the argument but now it doesn't start the LSP server. If I saw it and assumed it's like `ruby-lsp` but with debugging output, I'd be very confused to see it just exits after indexing.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -42,3 +42,7 @@ jobs:
 
       - name: Run tests
         run: bundle exec rake
+
+      - name: Run index troubleshooting tool
+        if: matrix.os != 'windows-latest'
```
Comment (andyw8) [inline .github/workflows/ci.yml:47]: I could not figure how why, but on Windows this fails with `bundler: not executable`.

**Review by vinistock — APPROVED** [review]

**Pull request merged**